### PR TITLE
prometheus-mpd-exporter: init at 0.1.0

### PR DESCRIPTION
**Pull request by matthiasbeyer**
###### Motivation for this change

Wrote this today and would find it useful if it were in nixpkgs. Others might be interested as well.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

I don't know how to properly test the service, but I have a similar service already running on two of my machines.

### CODE REVIEW

File: nixos/modules/services/monitoring/prometheus/exporters/mpd.nix [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+with lib;
+let
+  cfg = config.services.prometheus.exporters.mpd;
+in
+{
+  port = 9123;
+  extraOpts = {
+    port = mkOption {
+      type = types.port;
+      default = "9123";
```
Comment (mweinelt) [inline nixos/modules/services/monitoring/prometheus/exporters/mpd.nix:12]: https://github.com/prometheus/prometheus/wiki/Default-port-allocations

File: pkgs/servers/monitoring/prometheus/mpd-exporter.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{ stdenv
+, rustPlatform
+, fetchgit
+, pkg-config
+, openssl
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "prometheus-mpd-exporter";
+  version = "v0.1.0";
```
Comment (SuperSandro2000) [inline pkgs/servers/monitoring/prometheus/mpd-exporter.nix:10]:
```suggestion
  version = "0.1.0";
```

File: pkgs/servers/monitoring/prometheus/mpd-exporter.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  src = fetchgit {
+    url = "https://git.beyermatthi.as/prometheus-mpd-exporter";
+    rev = version;
```
Comment (SuperSandro2000) [inline pkgs/servers/monitoring/prometheus/mpd-exporter.nix:14]:
```suggestion
    rev = "v${version}";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108342` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package blacklisted:
</summary>
<ul>
<li>tests.nixos-functions.nixos-test</li>
</ul>
</details>
<details>
<summary>
1 package built:
</summary>
<ul>
<li>prometheus-mpd-exporter</li>
</ul>
</details>

File: pkgs/servers/monitoring/prometheus/mpd-exporter.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  nativeBuildInputs = [ pkg-config openssl ];
+
+  meta = with stdenv.lib; {
+    description = "Export mpd metrics to prometheus";
+    homepage = "https://git.beyermatthi.as/prometheus-mpd-exporter";
+    license = with licenses; [ gpl2.0 ];
+    maintainers = with maintainers; [ matthiasbeyer ];
+  };
+}
+
```
Comment (SuperSandro2000) [inline pkgs/servers/monitoring/prometheus/mpd-exporter.nix:29]: Please remove the extra 2 new lines.

File: pkgs/servers/monitoring/prometheus/mpd-exporter.nix [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  src = fetchgit {
+    url = "https://git.beyermatthi.as/prometheus-mpd-exporter";
+    rev = "$v{version}";
+    sha256 = "0nizmz29b64cywamfhfwwjbsfb3n9nb2dndl35li9z4v67lb5khi";
+  };
+
+  cargoSha256 = "0snrfcjawla9ra40x22w5zrr8xnsyhlyj21di8gp5wzf0yj1a9n1";
+
+  nativeBuildInputs = [ pkg-config openssl ];
```
Comment (SuperSandro2000) [inline pkgs/servers/monitoring/prometheus/mpd-exporter.nix:20]: if it links against openssl it need to go to buildInputs.

File: nixos/modules/services/monitoring/prometheus/exporters/mpd.nix [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+{
+  port = 9806;
+  extraOpts = {
+    port = mkOption {
+      type = types.port;
+      default = "9806";
+      description = ''
+        Port to bind to
+      '';
+    };
```
Comment (WilliButz) [inline nixos/modules/services/monitoring/prometheus/exporters/mpd.nix:16]:
```suggestion
```
`port` is a default option, it is already defined in `exporters.nix`

**Review by WilliButz — CHANGES_REQUESTED** [review]

**Comment by riotbib** [comment]
I am interested in this software being in nixpkgs.

May someone fill me in, why this merge request was closed?

**Comment by matthiasbeyer** [comment]
Because I don't care enough. I compile it myself in my `configuration.nix` and that's it.

**Pull request closed without merging**